In [26]:
# imported libraries which are used in this Project
import pandas as pd
import matplotlib.pyplot as plt 
import seaborn as sns

In [27]:
# reading the data from datasets
data = pd.read_csv('Advertising.csv')

In [29]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 10 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Daily Time Spent on Site  1000 non-null   float64
 1   Age                       1000 non-null   int64  
 2   Area Income               1000 non-null   float64
 3   Daily Internet Usage      1000 non-null   float64
 4   Ad Topic Line             1000 non-null   object 
 5   City                      1000 non-null   object 
 6   Male                      1000 non-null   int64  
 7   Country                   1000 non-null   object 
 8   Timestamp                 1000 non-null   object 
 9   Clicked on Ad             1000 non-null   int64  
dtypes: float64(3), int64(3), object(4)
memory usage: 78.2+ KB


### Preprocess the Data

In [30]:
# step:1 check for missing values
data.isnull().sum()

Daily Time Spent on Site    0
Age                         0
Area Income                 0
Daily Internet Usage        0
Ad Topic Line               0
City                        0
Male                        0
Country                     0
Timestamp                   0
Clicked on Ad               0
dtype: int64

In [31]:
# step:2 check for duplicate values
data.duplicated().sum()

0

In [32]:
# Deleting the unwanted columns from the dataset
data.drop('Ad Topic Line',axis=1,inplace=True)
data.drop('City',axis=1,inplace=True)
data.drop('Country',axis=1,inplace=True)
data.drop('Timestamp',axis=1,inplace=True)

In [33]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 6 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Daily Time Spent on Site  1000 non-null   float64
 1   Age                       1000 non-null   int64  
 2   Area Income               1000 non-null   float64
 3   Daily Internet Usage      1000 non-null   float64
 4   Male                      1000 non-null   int64  
 5   Clicked on Ad             1000 non-null   int64  
dtypes: float64(3), int64(3)
memory usage: 47.0 KB


In [34]:
# we have stored variable 'Clicked on Ad' in y variable
y = data['Clicked on Ad']

In [35]:
# Now we drop the dependant variable from data
data.drop('Clicked on Ad',axis=1,inplace=True)
x = data

In [36]:
#We have changed datatype from float to integer
x['Daily Time Spent on Site'] = x['Daily Time Spent on Site'].astype(int)
x['Daily Internet Usage'] = x['Daily Internet Usage'].astype(int)
x['Area Income'] = x['Area Income'].astype(int)

In [37]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype
---  ------                    --------------  -----
 0   Daily Time Spent on Site  1000 non-null   int32
 1   Age                       1000 non-null   int64
 2   Area Income               1000 non-null   int32
 3   Daily Internet Usage      1000 non-null   int32
 4   Male                      1000 non-null   int64
dtypes: int32(3), int64(2)
memory usage: 27.5 KB


### Spliting Data Set into Train Test Split

In [38]:
# we have imported train text split from sci-kit learn library
from sklearn.model_selection import train_test_split

In [39]:
x_train, x_test, y_train, y_test = train_test_split(x,y,test_size=0.2)

In [40]:
x_train.shape

(800, 5)

In [41]:
y_test.shape

(200,)

### Importing logistic Regression Algorithm

In [42]:
# We have imported logistic Regression from Sci-kit learn library
from sklearn.linear_model import LogisticRegression

In [43]:
lr = LogisticRegression()

In [44]:
lr.fit(x_train,y_train)

LogisticRegression()

In [45]:
lr.score(x_test,y_test)

0.93

### Conclusion
### Logistic regression gives good results on this Data set. logistic regression score is 0.93 or 93%